pkg/registry/core/service/storage/rest_test.go

```diff
@@ -948,202 +706,100 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
 	if created {
 		t.Errorf("expected not created")
 	}
-	if storage.serviceNodePorts.Has(30020) {
+	if storage.serviceNodePorts.Has(int(svc.Spec.Ports[0].NodePort)) {
 		t.Errorf("unexpected side effect: NodePort allocated")
 	}
-	if e, a := "", registry.UpdatedID; e != a {
-		t.Errorf("Expected %q, but got %q", e, a)
-	}
 
 	// Test dry run update request external name to cluster ip
-	_, _, err = storage.Update(ctx, svc.Name, rest.DefaultUpdatedObjectInfo(&api.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:            svc.Name,
-			ResourceVersion: svc.ResourceVersion},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeClusterIP,
-			ClusterIP:       "1.2.3.4",
-			ClusterIPs:      []string{"1.2.3.4"},
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+	new2 := svc.DeepCopy()
+	svctest.SetTypeClusterIP(new2)
+	_, _, err = storage.Update(ctx, svc.Name, rest.DefaultUpdatedObjectInfo(new2),
+		rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
 	if err != nil {
 		t.Fatalf("Expected no error: %v", err)
 	}
-	if storage.serviceIPAllocatorsByFamily[storage.defaultServiceIPFamily].Has(net.ParseIP("1.2.3.4")) {
+	if storage.serviceIPAllocatorsByFamily[storage.defaultServiceIPFamily].Has(net.ParseIP(svc.Spec.ClusterIP)) {
 		t.Errorf("unexpected side effect: ip allocated")
 	}
 
 	// Test dry run update request remove node port
-	obj, err = storage.Create(ctx, &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo2", ResourceVersion: "1", Namespace: metav1.NamespaceDefault},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeNodePort,
-			ClusterIP:       "1.2.3.5",
-			ClusterIPs:      []string{"1.2.3.5"},
-			Ports: []api.ServicePort{{
-				NodePort:   30020,
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	obj, err = storage.Create(ctx, svctest.MakeService("foo2", svctest.SetTypeNodePort), rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Expected no error: %v", err)
 	}
 	svc = obj.(*api.Service)
+	if !storage.serviceIPAllocatorsByFamily[storage.defaultServiceIPFamily].Has(net.ParseIP(svc.Spec.ClusterIP)) {
```

Comment:
this test is not need. We have already covered this in L726

Reply:
same thing as in https://github.com/kubernetes/kubernetes/pull/103379/files#r662225171